XRTK-Core/Packages/com.xrtk.core/Extensions/ArrayExtensions.cs

```diff
@@ -37,12 +37,43 @@ public static bool IsValidArray(this Array array)
         /// <param name="array">The array to extend</param>
         /// <param name="newItem">The item to add to the array</param>
         /// <returns></returns>
-        public static T[] AddItem<T>(this T[] array, T newItem, int insertAtIndex = 0)
+        public static T[] AddItem<T>(this T[] array, T newItem)
         {
+            // Initialise the array if it is null
+            if (array == null)
+            {
+                array = new T[0];
```

Comment:
Take a look at my changed in [455](https://github.com/XRTK/XRTK-Core/pull/455/files?file-filters%5B%5D=.cs&hide-deleted-files=true#diff-d33dfd0c48086e9a05bb9888d60a437bR45)
```suggestion
                return new[] { newItem };
```